src/Findologic/Response/Xml21ResponseParser.php

```diff
@@ -184,4 +187,49 @@ private function isFilterSet(array $params, string $name): bool
     {
         return isset($params[$name]) && !empty($params[$name]);
     }
+
+    public function getFiltersWithSmartSuggestBlocks(
+        FiltersExtension $flFilters,
+        array $flBlocks,
+        array $params
+    ): FiltersExtension {
+        $hasCategoryFilter = $hasVendorFilter = false;
+
+        foreach ($flFilters->getFilters() as $filter) {
+            if ($filter instanceof CategoryFilter) {
+                $hasCategoryFilter = true;
+            }
+            if ($filter instanceof VendorImageFilter) {
+                $hasVendorFilter = true;
+            }
+        }
+
+        if (!$hasVendorFilter || !$hasCategoryFilter) {
+            if (!$hasCategoryFilter && array_key_exists('cat', $flBlocks) && $this->isFilterSet($params, 'cat')) {
+                $display = $flBlocks['cat'];
+                $value = $params['cat'];
+
+                $customFilter = new CategoryFilter('cat', $display);
+                $filterValue = new FilterValue($value, $value, 'cat');
+                $customFilter->addValue($filterValue);
+                $customFilter->setHidden(true);
+
+                $flFilters->addFilter($customFilter);
```

Comment:
Create a method like `buildHiddenFilter`, which returns a `Filter`. Use this method to build the hidden filter for cat/vendor.